test/index.test.js

```diff
@@ -296,26 +292,82 @@ describe('Anvil API Client', function () {
             } = options
 
             expect(method).to.eql('POST')
-            expect(headers).to.eql({}) // node-fetch will add appropriate header
-            expect(body).to.be.an.instanceof(FormData)
-            expect(signal).to.be.an.instanceof(AbortSignal)
-            expect(
-              FormData.prototype.append.withArgs(
-                'map',
-                JSON.stringify({ 1: ['variables.aNested.file'] }),
-              ),
-            ).calledOnce
+            if ($.isBase64) {
+              expect(headers).to.eql({
+                'Content-Type': 'application/json',
+              })
+              // Vars are untouched
+              expect(JSON.parse(body).variables).to.eql($.variables)
+            } else {
+              expect(headers).to.eql({}) // node-fetch will add appropriate header
+              expect(body).to.be.an.instanceof(FormData)
+              expect(signal).to.be.an.instanceof(AbortSignal)
+              expect(
+                FormData.prototype.append.withArgs(
+                  'map',
+                  JSON.stringify({ 1: ['variables.aNested.file'] }),
+                ),
+              ).calledOnce
+            }
+          })
+
+          context('file is a Buffer', function () {
+            def('variables', () => ({
+              aNested: {
+                file: Buffer.from(''),
+              },
+            }))
+
+            it('creates a FormData and appends the files map', function () {
+              client.requestGraphQL({ query, variables: $.variables }, clientOptions)
+            })
+          })
+
+          context('file is a Stream', function () {
+            def('variables', () => {
+              const file = fs.createReadStream(path.join(assetsDir, 'dummy.pdf'))
```

Comment:
Does this provide the filename / mime? Will this fail asking for a filename / mime when a stream without that info is provided? I assume it would be specified just like the base64 one below?

Reply:
Great question, but no we are ok. As I [wrote in our docs](https://github.com/anvilco/node-anvil/pull/20/files#diff-04c6e90faac2675aa89e2176d2eec7d8R131), `form-data` will be able to automatically suss out the filename and mime stuff from several "common" types of streams that it deals with:
> Form-Data can recognize and fetch all the required information from common types of streams (fs.readStream, http.response and mikeal's request), for some other types of streams you'd need to provide "file"-related information manually

What happens is that it finds all that info in the "stream"-like thing's properties and it ends up in the multipart form head junk...and on the server side, `GraphQL Upload` will leverage it.

Our API supports `form-data` API's allowance for the manually passing/specifying of filename and mime stuff for these "non common" scenarios via the `options` param that I just referenced here, so users are good to go for all scenarios.

Basically, if `form-data` supports it (which is the only way we can support it), then we can also support it...but sometimes the User has to provide the `options` param depending on what stream-like thing they're going to use.

I've tried this against the API for reals and logged out a bunch of scenarios and it's all good.

The base64 case is obviously different b/c from the framework's point of view this is just your average JSON with some string value.